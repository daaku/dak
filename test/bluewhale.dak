(import [:node:fs/promises [opendir]]
        [:node:path [join dirname]]
        [:node:url [fileURLToPath]]
        [:bun:test [test expect]]
        [:../src/transpiler.dak [transpileStr]])

(const root (-> import.meta.url fileURLToPath (join "../__data__/bluewhale")))

(fn sanitize-stack [stack]
  (and stack (-> stack (.replace (str (dirname root) "/") ""))))

(const AsyncFunction (. (fn@ []) :constructor))

(for@ [ent @(opendir root)]
  (test ent.name
    (fn@ []
      (let [out (try
                   @((-> (join root ent.name) Bun.file .text await
                         (transpileStr {:source ent.name :sourcemap :inline})
                         (. :code)
                         AsyncFunction))
                   (catch e `${e.message}\n\n--\n\n${(sanitize-stack e.stack)}`))]
        (-> out expect .toMatchSnapshot)))))
