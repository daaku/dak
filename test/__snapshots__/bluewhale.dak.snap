// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`001-js-in-dak.dak 1`] = `
""hello" is not a function. (In '"hello"()', '"hello"' is "hello")

--

TypeError: "hello" is not a function. (In '"hello"()', '"hello"' is "hello")
    at <anonymous> (:3:20)
    at anonymous (:1:26)
    at <anonymous> (test/bluewhale.dak:0:632)"
`;
